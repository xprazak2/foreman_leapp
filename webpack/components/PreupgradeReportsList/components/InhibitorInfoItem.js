import React from 'react';
import PropTypes from 'prop-types';
import { ListView, Icon } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

import { isInhibitor } from '../../PreupgradeReports/PreupgradeReportsHelpers';
import InfoItem from './InfoItem';
import EmptyInfoItem from './EmptyInfoItem';

const InhibitorInfoItem = ({ entry }) => {
  const key = `inhibitor-${entry.id}`;

  if (isInhibitor(entry)) {
    return (
      <InfoItem itemId={key} tooltipText={__('This issue inhibits the upgrade.')}>
        <Icon type="fa" name="icon-exclamation-sign" />
      </InfoItem>
    )
  }
  return <EmptyInfoItem key={key} entry={entry} attr="flags" />
}

InhibitorInfoItem.propTypes = {
  entry: PropTypes.object.isRequired,
};

export default InhibitorInfoItem;
