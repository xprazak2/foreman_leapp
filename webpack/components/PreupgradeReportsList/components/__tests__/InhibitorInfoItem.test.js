import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import InhibitorInfoItem from '../InhibitorInfoItem';

const fixtures = {
  'should render when inhibitor': {
    entry: { title: 'no title', severity: 'low', id: 5, preupgradeReportId: 7, flags: ['inhibitor'] },
  },
  'should render when not inhibitor': {
    entry: { title: 'no title', severity: 'low', id: 5, preupgradeReportId: 7, flags: [] },
  },
};

describe('InhibitorInfoItem', () =>
  testComponentSnapshotsWithFixtures(InhibitorInfoItem, fixtures));
