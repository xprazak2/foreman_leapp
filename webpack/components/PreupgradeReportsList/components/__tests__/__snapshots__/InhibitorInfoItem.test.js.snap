// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InhibitorInfoItem should render when inhibitor 1`] = `
<InfoItem
  itemId="inhibitor-5"
  tooltipText="This issue inhibits the upgrade."
>
  <Icon
    name="icon-exclamation-sign"
    type="fa"
  />
</InfoItem>
`;

exports[`InhibitorInfoItem should render when not inhibitor 1`] = `
<EmptyInfoItem
  attr="flags"
  entry={
    Object {
      "flags": Array [],
      "id": 5,
      "preupgradeReportId": 7,
      "severity": "low",
      "title": "no title",
    }
  }
  key="inhibitor-5"
/>
`;
